### ISSUE TITLE
ML.NET Training Failure

### ISSUE THREAD

**Issue by acsamusements** [dotnet/machinelearning-modelbuilder#2779]
**System Information (please complete the following information):**
 - Model Builder or CLI Version: ML.NET 17.17.0
 - Visual Studio Version (if applicable): Visual Studio 2022 17.7.4 64-bit

**Describe the bug**
- On which page of the tutorial did you run into an issue (URL):
- Clear description of the problem:
- After Clicking "Start Training" few seconds later error

**To Reproduce**

**Expected behavior**
Full training, but don't get through that without error

**Screenshots**
![image](https://github.com/dotnet/machinelearning-modelbuilder/assets/16455904/53038ad2-3a4c-48e6-a6ac-8396083d8a61)

**Additional context**
Model Builder Error Top half
arget 4294967295 is out of bounds.
Exception raised from nll_loss_out_frame at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\LossNLL.cpp:230 (most recent call first):
00007FFEB155AD0200007FFEB155ACA0 c10.dll!c10::Error::Error [<unknown file> @ <unknown line number>]
00007FFEB1523FB500007FFEB1523F40 c10.dll!c10::IndexError::IndexError [<unknown file> @ <unknown line number>]
00007FFD9BB6C24400007FFD9BB674C0 torch_cpu.dll!at::native::multi_margin_loss_cpu_out [<unknown file> @ <unknown line number>]
00007FFD9BB6EDC400007FFD9BB674C0 torch_cpu.dll!at::native::multi_margin_loss_cpu_out [<unknown file> @ <unknown line number>]
00007FFD9BB719D400007FFD9BB71970 torch_cpu.dll!at::native::structured_nll_loss_forward_out_cpu::impl [<unknown file> @ <unknown line number>]
00007FFD9C6E53CE00007FFD9C6CB180 torch_cpu.dll!at::cpu::zero_ [<unknown file> @ <unknown line number>]
00007FFD9C6987FE00007FFD9C665CA0 torch_cpu.dll!at::cpu::bucketize_outf [<unknown file> @ <unknown line number>]
00007FFD9C14177D00007FFD9C0DE640 torch_cpu.dll!at::_ops::zeros_out::redispatch [<unknown file> @ <unknown line number>]
00007FFD9C36AB3300007FFD9C36AA90 torch_cpu.dll!at::_ops::nll_loss_forward::redispatch [<unknown file> @ <unknown line number>]
00007FFD9D71213100007FFD9D536BC0 torch_cpu.dll!torch::autograd::GraphRoot::apply [<unknown file> @ <unknown line number>]
00007FFD9D6DEB5200007FFD9D536BC0 torch_cpu.dll!torch::autograd::GraphRoot::apply [<unknown file> @ <unknown line number>]
00007FFD9C2F58E000007FFD9C2F5710 torch_cpu.dll!at::_ops::nll_loss_forward::call [<unknown file> @ <unknown line number>]
00007FFD9BB7219F00007FFD9BB72120 torch_cpu.dll!at::native::nll_loss [<unknown file> @ <unknown line number>]
00007FFD9C9A85B200007FFD9C9A4610 torch_cpu.dll!at::compositeimplicitautograd::where [<unknown file> @ <unknown line number>]
00007FFD9C98D50D00007FFD9C95FAC0 torch_cpu.dll!at::compositeimplicitautograd::broadcast_to [<unknown file> @ <unknown line number>]
00007FFD9C4222D700007FFD9C422110 torch_cpu.dll!at::_ops::nll_loss::call [<unknown file> @ <unknown line number>]
00007FFD9BB72DE700007FFD9BB72210 torch_cpu.dll!at::native::nll_loss_nd [<unknown file> @ <unknown line number>]
00007FFD9C9A85E200007FFD9C9A4610 torch_cpu.dll!at::compositeimplicitautograd::where [<unknown file> @ <unknown line number>]
00007FFD9C98D57D00007FFD9C95FAC0 torch_cpu.dll!at::compositeimplicitautograd::broadcast_to [<unknown file> @ <unknown line number>]
00007FFD9C1A643700007FFD9C1A6270 torch_cpu.dll!at::_ops::nll_loss_nd::call [<unknown file> @ <unknown line number>]
00007FFD9BB7062F00007FFD9BB70340 torch_cpu.dll!at::native::cross_entropy_loss [<unknown file> @ <unknown line number>]
00007FFD9C9A68A100007FFD9C9A4610 torch_cpu.dll!at::compositeimplicitautograd::where [<unknown file> @ <unknown line number>]
00007FFD9C98D60200007FFD9C95FAC0 torch_cpu.dll!at::compositeimplicitautograd::broadcast_to [<unknown file> @ <unknown line number>]
00007FFD9C3FF8AF00007FFD9C3FF6C0 torch_cpu.dll!at::_ops::cross_entropy_loss::call [<unknown file> @ <unknown line number>]
00007FFE98059A7400007FFE980598C0 LibTorchSharp.DLL!THSNN_cross_entropy [<unknown file> @ <unknown line number>]
00007FFE64D10267 <unknown symbol address> !<unknown symbol> [<unknown file> @ <unknown line number>]

Bottom Half
 at TorchSharp.torch.CheckForErrors()
   at TorchSharp.Modules.CrossEntropyLoss.forward(Tensor input, Tensor target)
   at Microsoft.ML.TorchSharp.NasBert.NasBertTrainer`2.NasBertTrainerBase.RunModelAndBackPropagate(Tensor& inputTensor, Tensor& targetsTensor)
   at Microsoft.ML.TorchSharp.TorchSharpBaseTrainer`2.TrainerBase.TrainStep(IHost host, DataViewRowCursor cursor, ValueGetter`1 labelGetter, List`1& inputTensors, List`1& targets)
   at Microsoft.ML.TorchSharp.TorchSharpBaseTrainer`2.TrainerBase.Train(IHost host, IDataView input)
   at Microsoft.ML.TorchSharp.TorchSharpBaseTrainer`2.Fit(IDataView input)
   at Microsoft.ML.Data.EstimatorChain`1.Fit(IDataView input)
   at Microsoft.ML.AutoML.SweepablePipelineRunner.Run(TrialSettings settings)
   at Microsoft.ML.AutoML.SweepablePipelineRunner.RunAsync(TrialSettings settings, CancellationToken ct)
   at Microsoft.ML.AutoML.AutoMLExperiment.<RunAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ML.ModelBuilder.AutoMLService.TextClassificationExperiment.<ExecuteAsync>d__13.MoveNext() in /_/src/Microsoft.ML.ModelBuilder.AutoMLService/Experiments/TextClassificationExperiment.cs:line 126
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ML.ModelBuilder.AutoMLEngine.<StartTrainingAsync>d__21.MoveNext() in /_/src/Microsoft.ML.ModelBuilder.AutoMLService/AutoMLEngineService/AutoMLEngine.cs:line 169
[SMTPLOGS-LWLGG5.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/12833843/SMTPLOGS-LWLGG5.txt)

**Comment by LittleLittleCloud** [comment]
Looks like it's a duplicate of https://github.com/dotnet/machinelearning-modelbuilder/issues/2369

**Comment by acsamusements** [comment]
Other than it being the latest version, yes it does look like it is the same issue.  My bad I search but, didn't come up anything.  If necessary you can close this one and refer it the other tickets.

**Comment by LittleLittleCloud** [comment]
@acsamusements That issue has been resolved in January,, This might be a regress.

Would you mind sharing your dataset with us, or check if the label in your dataset contains `0`?

**Comment by acsamusements** [comment]
@LittleLittleCloud Seems to issues with Blanks or null values.  Dataset had a row at the beginning and end the made the particular columns have blank/null values.  After removing those row and a completely empty row at the end seems to have fixed that issue.

**Comment by luisquintanilla** [comment]
Thanks @LittleLittleCloud. Seems like the issue is resolved. Closing this issue.
